pkg/minikube/cluster/cluster.go

```diff
@@ -78,6 +78,20 @@ func StartHost(api libmachine.API, config MachineConfig) (*host.Host, error) {
 	}
 
 	s, err := h.Driver.GetState()
+
+	// This is the name of the driver passed in
+	// through a flag, env, or config file (if any)
+	driverSpecified := config.VMDriver
+
+	// This is the name of the driver that is currently used with minikube
+	currentDriver := h.Driver.DriverName()
+
+	if driverSpecified != "" && driverSpecified != currentDriver {
```

Comment:
I like the idea. Could we turn this into a general validation here, stuff like CPUs, memory, etc. are also confusing in that they can't be changed with another start invocation.

I guess you could turn this into a loop over a list of flags that require a restart.